### ISSUE TITLE
Multiple reading of the same data

### ISSUE THREAD

**Issue by salamander2** [RPG-code/myRPGGame#2]
The same data files are being read each time a new character is generated. They should just be read once and the data stored in the program.

**Comment by ShadowMole** [comment]
Okay, so this was the first time that I have ever read data in from another file.  I really didn't know what I was doing with it.

**Comment by salamander2** [comment]
Hey, no problem. That's what this is all about -- learning new things. You've got a bunch of things to learn at the beginning (git, github, ...) and I'll have some more things to learn as we get further in.

I did some work, but stopped a few days ago. I think that the next thing to do is to get rid of the code that duplicates **"story.txt"** in **Character.java.**  It's really not good style to hard code things in and then also read them from a file. If you want to change the .txt file, you then also have to change the code.

What are you working on? Are you interested in fixing this part?  *I haven't had time yet to go through the whole thing and get to the final problem where you were running out of space due to huge arrays.*

**Comment by salamander2** [comment]
Oh, I forgot to say, you can see what I did with Magic and Race. Either file can be edited and the program will reflect that. I've also implemented comments in the files.

And the code still compiles. I won't upload it if it doesn't compile.   Let me know what you want to work on or learn.

**Comment by ShadowMole** [comment]
I think we should try to optimize the code that is current;y there.  I haven't had time to really work on this recently because of vacation and meeting friends because we all on break from different colleges.  I did have time today to look though some your changes.  I'm not sure how some of the really work, but I really only glanced at the code.  In the next two or three days, I will be working on the code.

I think I understand GitHub now now that I am actually using it.  As for what I would to work on, well there is a lot of things.

1.  I would like to implement a movement system.  This will be on three levels.  Characters should be able to move within rooms, between rooms, and between areas.  Changing rooms and areas should be mostly the same, and I should be able to implement that.  I am not sure how to let players move inside of a room.

2.  I would like to make GUI for this program, simply because it will make user interaction so much easier.  I don't know how to make a GUI because that will be covered in my spring semester, but I would really like to know how to do it.

3.  Creating different kinds of Items.  So right now, there is the Armor class and the Weapon class.  I really just made those because I wanted to finish the character creation.  In the end both of those should extend a generic Item class.  I plan to implement several other Item subclasses as well.

4.  I would like to work on a creating a parent class Skill.  There will be three subclasses: NonCombat, WeaponSkill, and Spell.  This may take a little to work out.

5.  I would like to create an Enemy class.  I would like for enemies to be randomly generated.  I need to work out how I would like this to be implemented though.

6.  I need to figure out how each of the characters takes a turn.

7.  I would like to optimize all of the code that is implemented and test it as we go.  I do not want to leave bugs in the code.  This also applies to what we currently have.

So those are the goals that I can remember right now.   I may have forgotten one or two.  Beyond what I have mentioned there isn't much that I want to learn, but that is only because I don't know what is out there.  I can pick things up pretty fast and I will ask a lot of questions as I start analyzing your code, so feel free to code as well as you can.  I don't know if you can learn anything from me though.

You really don't have to work on the Analysis class.  It doesn't provide anything for the end result.  I only created that method because I like to work with and analyze numbers in my spare time.  I would rather focus on the end result.  However, if you want to work on it go ahead.

Pick whatever sounds the most interesting and let me know tomorrow.  I hope to have something new to contribute by Thursday or Friday.

**Comment by salamander2** [comment]
Wow, you have lots of plans for this. I've just uploaded my TextAdventure Game that does some of these things, especially movement and items (# 1,  #3 - kind of). I also know how to do # 2 and #6 and can tell you this when we get to it.

I'd like to work on #7 still: optimizing the code. I think that continuing to fix up the Character and FileRead classes would  be best to do first. 

- What questions do you have when you look at my code?
- :question: Could you fix up the "story.txt" part of the code, similar to how I did the race.txt ? I ended up having to add an extra line of description to race.txt, but you wouldn't need to do that for story.txt
- I will work on the modStr and adding the various buff modifications ending up with modded_xxx
- I'll also change the part where the extra 20 points that can be added to make it more user friendly.

Next I'll need to examine the Weapon and Armor classes. You have two completely different constructors: e.g Weapon(int level) and Weapon(String type). Based on what objects are, this is most likely a non-good way of doing it. Some weapons are created based on a level number and some are created based on a String, so this should probably be changed.  

I think that what you're doing is making a master list of weapons and alternate names for them. If so, then this list should be static, so that there is just one copy of them for the whole game. (And we need to add some comments to say things like this in the code).

**Comment by ShadowMole** [comment]
I agree that working with the current code would be best.  I will work on story.txt either today or tomorrow.  

As for the Weapon constructors, the first one is for the weapon that are randomly generated within the rooms.  This is based on what area the room is in and helps to decide the quality of the weapon.  The second one is to create the beginning weapons for characters.  In this one the quality is fixed, but everything else is based on the Background of the character and is pulled from story.txt.  The Armor constructors are the same.

That is why they are different, however, there probably is a better way of doing this.  A master list would also be a good idea.

**Comment by ShadowMole** [comment]
I uploaded up much smaller game that I have completed.  It started as a class assignment so some of the code was given to us, Such as the Enum and the Case and Switch portions.  Most of everything else has been either editted or added by me, so it should be able to show you what I know how to do.

**Comment by salamander2** [comment]
Thanks. I've glanced at it, but will read over it in more detail today.

-----
Switch is simple. It's like a series of If statements, but all using the same variable. It's simpler and easier to prevent errors. See http://quarkphysics.ca/ICS3U1/unit2/if_switch.htm#switch

Enum: yeah. Well, at some point we can look into if when we need it.

-----

Hey, you don't need to be copying all of my changes to your own repository (unless you want to, of course). You should be able to edit things on this one. I've set it up as a team thing so that both of us can work on it.

Try and edit the Readme.MD on the main page to see if you can edit code here. 

Also, don't make a new branch for every change. What you want is a new commit for every change. Branches are supposed to get merged back to the main code once whatever the problem is has been fixed.

**Comment by salamander2** [comment]
Hi. 
I'm feeling my interest fading away. I'm getting a bit busy, and also there is a lot that I don't understand. I don't see how you can get to a level 6 with like 87 barracks and 22 guard rooms. Where are they all located? and then each room can have many many chests, weapons racks etc. It's a bit overwhelming and I don't know how it will work. 

I can do a bit more fixing up and set up player turns.

I think that there must be a better way to communicate. This is not working that well.  I'm also going to be getting busier next week :(

**Comment by ShadowMole** [comment]
It's fine if you want to drop the project.  I haven't really had the time to work on the code because I have been catching up with friends and my brother because we are all home from college on break right now.  I will be picking it up again when my new semester starts at the end of January, but I'm not sure how much I will be doing until then.

Also, this is a very ambitious project, and I am sorry if this is overwhelming.  I just don't like doing small projects unless I am leaning a new concept.  I have a tendency to make my personal projects overly complicated.

Anyway thank you for what help you were able to provide.  If you would like to stop working on it that is fine and I completely understand.
